develop/blockchains/deployment/build-deterministic-runtime.md

```diff
@@ -0,0 +1,138 @@
+
+The [`srtool-cli`](https://github.com/chevdor/srtool-cli){target=\_blank} package is a command-line utility written in Rust that installs an executable program called `srtool`. This program simplifies the interactions with the `srtool` Docker container.
+
+Over time, the tooling around the `srtool` Docker image has expanded to include the following tools and helper programs:
+
+- [**`srtool-cli`**](https://github.com/chevdor/srtool-cli){target=\_blank} - provides a command-line interface to pull the srtool Docker image, get information about the image and tooling used to interact with it, and build the runtime using the `srtool` Docker container
+- [**`subwasm`**](https://github.com/chevdor/subwasm){target=\_blank} - provides command-line options for working with the metadata and Wasm runtime built using srtool. The `subwasm` program is also used internally to perform tasks in the `srtool` image
+- [**`srtool-actions`**](https://github.com/chevdor/srtool-actions){target=\_blank} - provides GitHub actions to integrate builds produced using the `srtool` image with your GitHub CI/CD pipelines
+- [**`srtool-app`**](https://gitlab.com/chevdor/srtool-app){target=\_blank} - provides a simple graphical user interface for building the runtime using the `srtool` Docker image
+
+## Prepare the Environment
+
+It is recommended to install the `srtool-cli` program to work with the Docker image using a simple command-line interface.
+
+To prepare the environment:
+
+1. Verify that Docker is installed by running the following command:
+
+    ```bash
+    docker --version
+    ```
+
+    If Docker is installed, the command will display version information:
+
+    --8<-- 'code/develop/parachain-devs/runtime-development/build-deterministic-runtime/deterministic-runtime-1.html'
+
+2. Install the `srtool` command-line interface by running the following command:
+
+    ```bash
+    cargo install --git https://github.com/chevdor/srtool-cli
+    ```
+
+3. View usage information for the `srtool` command-line interface by running the following command:
+
+    ```bash
+    srtool help
+    ```
+
+4. Download the latest `srtool` Docker image by running the following command:
+
+    ```bash
+    srtool pull
+    ```
+
+## Start a Deterministic Build
+
+After preparing the environment, the Wasm runtime can be compiled using the `srtool` Docker image.
+
+To build the runtime, you need to open your Polkadot SDK-based project in a terminal shell and run the following command:
+
+```bash
+srtool build --app --package INSERT_RUNTIME_PACKAGE_NAME --runtime-dir INSERT_RUNTIME_PATH 
+```
+
+- The name specified for the `--package` should be the name defined in the `Cargo.toml` file for the runtime
+- The path specified for the `--runtime-dir` should be the path to the `Cargo.toml` file for the runtime. For example:
+
+    ```plain
+    node/
+    pallets/
+    runtime/
+        lib.rs
+        Cargo.toml # INSERT_RUNTIME_PATH should be the path to this file
+    ...
+    ```
```

Comment:
```suggestion
    ```plain
    node/
    pallets/
    runtime/
    ├──lib.rs
    └──Cargo.toml # INSERT_RUNTIME_PATH should be the path to this file
    ...
    ```
```